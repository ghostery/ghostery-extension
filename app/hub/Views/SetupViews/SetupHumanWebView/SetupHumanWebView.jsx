/**
 * Setup Human Web View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { ToggleCheckbox } from '../../../../shared-components';

/**
 * A Functional React component for rendering the Setup Human Web View
 * @return {JSX} JSX for rendering the Setup Human Web View of the Hub app
 * @memberof HubComponents
 */
const SetupHumanWebView = props => (
	<div className="row align-center">
		<div className="columns small-12 large-10">
			<div className="SetupHumanWeb">
				<div className="SetupHumanWeb__header" dangerouslySetInnerHTML={{ __html: t('hub_setup_humanweb_header') }} />
				<div className="flex-container align-middle">
					<ToggleCheckbox
						checked={props.enableHumanWeb}
						onChange={props.changeHumanWeb}
					/>
					<span className="SetupHumanWeb__label" onClick={props.changeHumanWeb}>
						{ t('hub_setup_humanweb_label') }
					</span>
				</div>
			</div>
		</div>
	</div>
);

// PropTypes ensure we pass required props of the correct type
SetupHumanWebView.propTypes = {
	enableHumanWeb: PropTypes.bool.isRequired,
	changeHumanWeb: PropTypes.func.isRequired,
};

export default SetupHumanWebView;
