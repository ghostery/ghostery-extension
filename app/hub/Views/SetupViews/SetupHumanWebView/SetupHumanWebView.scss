/**
 * Setup Human Web View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Setup Human Web View
.SetupHumanWeb__header {
	color: #4a4a4a;
	font-size: 20px;
	line-height: 38px;
	margin: 15px 15px 60px;
}
.SetupHumanWeb__header a {
	color: #00aef0;
}
.SetupHumanWeb__label {
	color: #4a4a4a;
	font-size: 14px;
	cursor: pointer;
}
