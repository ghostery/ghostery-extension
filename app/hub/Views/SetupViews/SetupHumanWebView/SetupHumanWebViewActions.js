/**
 * Setup Human Web View Action creators
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { log, sendMessageInPromise } from '../../../utils';
import { SET_HUMAN_WEB } from '../../SetupView/SetupViewConstants';

export function setHumanWeb(actionData) {
	return function (dispatch) {
		return sendMessageInPromise(SET_HUMAN_WEB, actionData).then((data) => {
			dispatch({
				type: SET_HUMAN_WEB,
				data,
			});
		}).catch((err) => {
			log('setupBlocking Action setHumanWeb Error', err);
		});
	};
}
