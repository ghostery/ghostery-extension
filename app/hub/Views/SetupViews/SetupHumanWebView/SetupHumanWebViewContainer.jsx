/**
 * Setup Human Web View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SetupHumanWebView from './SetupHumanWebView';

/**
 * @class Implement the Setup Human Web View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class SetupHumanWebViewContainer extends Component {
	/**
	 * Lifecycle Event
	 */
	componentWillMount() {
		const title = t('hub_setup_page_title_humanweb');
		window.document.title = title;

		const { index, setup, sendMountActions } = this.props;
		this.props.actions.setSetupNavigation({
			activeIndex: index,
			hrefPrev: `/setup/${index - 1}`,
			hrefNext: `/setup/${index + 1}`,
			hrefDone: '/',
			textPrev: t('hub_setup_nav_previous'),
			textNext: t('hub_setup_nav_next'),
			textDone: t('hub_setup_exit_flow'),
		});

		if (sendMountActions) {
			const { enable_human_web } = setup;
			this.props.actions.setHumanWeb({ enable_human_web });
		}
	}

	/**
	* Function to handle toggling Human Web Opt-In
	*/
	_handleToggle = () => {
		const enable_human_web = !this.props.setup.enable_human_web;
		this.props.actions.setHumanWeb({ enable_human_web });
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Setup Human Web View of the Hub app
	 */
	render() {
		const childProps = {
			enableHumanWeb: this.props.setup.enable_human_web,
			changeHumanWeb: this._handleToggle,
		};
		return <SetupHumanWebView {...childProps} />;
	}
}

// PropTypes ensure we pass required props of the correct type
SetupHumanWebViewContainer.propTypes = {
	index: PropTypes.number.isRequired,
	actions: PropTypes.shape({
		setSetupNavigation: PropTypes.func.isRequired,
		setHumanWeb: PropTypes.func.isRequired,
	}).isRequired,
};

export default SetupHumanWebViewContainer;
