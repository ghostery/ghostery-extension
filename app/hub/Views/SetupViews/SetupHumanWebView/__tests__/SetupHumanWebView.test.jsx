/**
 * Setup Human Web View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SetupHumanWebView from '../SetupHumanWebView';

describe('app/hub/Views/SetupViews/SetupHumanWebView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('setup human web view is rendered correctly', () => {
			const initialState = {
				enableHumanWeb: true,
				changeHumanWeb: () => {},
			};

			const component = renderer.create(
				<SetupHumanWebView {...initialState} />
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				enableHumanWeb: true,
				changeHumanWeb: () => {},
			};

			const component = shallow(<SetupHumanWebView {...initialState} />);
			expect(component.find('.SetupHumanWeb').length).toBe(1);
			expect(component.find('.SetupHumanWeb__label').length).toBe(1);
		});
	});
});
