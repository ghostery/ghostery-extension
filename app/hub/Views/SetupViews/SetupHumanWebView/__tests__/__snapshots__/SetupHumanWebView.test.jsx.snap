// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/SetupViews/SetupHumanWebView component Snapshot tests with react-test-renderer setup human web view is rendered correctly 1`] = `
<div
  className="row align-center"
>
  <div
    className="columns small-12 large-10"
  >
    <div
      className="SetupHumanWeb"
    >
      <div
        className="SetupHumanWeb__header"
        dangerouslySetInnerHTML={
          Object {
            "__html": "hub_setup_humanweb_header",
          }
        }
      />
      <div
        className="flex-container align-middle"
      >
        <div
          className="ToggleCheckbox ToggleCheckbox--active"
          onClick={[Function]}
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </div>
        <span
          className="SetupHumanWeb__label"
          onClick={[Function]}
        >
          hub_setup_humanweb_label
        </span>
      </div>
    </div>
  </div>
</div>
`;
